ui/src/store/ConversationStore.ts

```
@@ -131,98 +135,123 @@ export class ConversationStore {
     return get(this.conversation);
   }
 
+  /**
+   * Public joining code for a conversation
+   */
   get publicInviteCode() {
-    if (this.data.privacy === Privacy.Public) {
-      const invitation: Invitation = {
+    if (this.data.privacy !== Privacy.Public)
+      throw new Error("Cannot generate public invite code for private conversation");
+
+    return Base64.fromUint8Array(
+      encode({
         created: this.created,
         networkSeed: this.data.id,
         privacy: this.data.privacy,
         progenitor: this.data.progenitor,
         title: this.title,
-      };
-      const msgpck = encode(invitation);
-      return Base64.fromUint8Array(msgpck);
-    } else {
-      return "";
-    }
+      } as Invitation),
+    );
   }
 
-  async inviteCodeForAgent(publicKeyB64: string) {
-    if (this.data.privacy === Privacy.Public) {
-      return this.publicInviteCode;
-    }
-    const proof = await this.relayStore.inviteAgentToConversation(
+  private async makeInviteCodeForAgent(publicKeyB64: string) {
+    if (this.data.privacy === Privacy.Public) return this.publicInviteCode;
+
+    const membraneProof = await this.relayStore.client.generateMembraneProofForAgent(
       this.data.id,
       decodeHashFromBase64(publicKeyB64),
     );
-    if (proof !== undefined) {
```

- throw errors from lower-level functions rather than passing up conditional return types